src/collections/safe_list.zig

```diff
@@ -75,21 +81,32 @@ pub fn SafeList(comptime T: type) type {
         }
 
         /// Add all the items in a slice to the end of this list.
-        pub fn appendSlice(self: *SafeList(T), gpa: Allocator, items: []const T) Slice {
+        pub fn appendSlice(self: *SafeList(T), gpa: Allocator, items: []const T) Range {
             const start_length = self.len();
             self.items.appendSlice(gpa, items) catch |err| exitOnOom(err);
-
-            return self.items.items[start_length..];
+            const end_length = self.len();
+            return Range{ .start = @enumFromInt(start_length), .end = @enumFromInt(end_length) };
         }
 
         /// Extend this list with all items generated by an iterator.
-        pub fn extendFromIter(self: *SafeList(T), gpa: Allocator, iter: anytype) Slice {
+        pub fn extendFromIter(self: *SafeList(T), gpa: Allocator, iter: anytype) Range {
             const start_length = self.len();
             while (iter.next()) |item| {
                 self.items.append(gpa, item) catch |err| exitOnOom(err);
             }
+            const end_length = self.len();
+            return Range{ .start = @enumFromInt(start_length), .end = @enumFromInt(end_length) };
+        }
 
-            return self.items.items[start_length..];
+        /// Convert a range to a slice
+        pub fn rangeToSlice(self: *const SafeList(T), range: *const Range) Slice {
```

Comment:
Should this take the range by value instead of pointer? `SafeRange` is only a small with two `Idx` values. Just a thought I had but I'm not sure. 

@bhansconnect do you have thoughts on this?